text/0000-a11y-routing.md

```diff
@@ -0,0 +1,91 @@
+1. When a browser loads a website, the screen reader starts to read the content of that page
+2. When the user navigates to a new page in that website, the screen reader starts to read the content of that page
+3. (repeat)
+
+Perhaps we could achieve this by adding a function to set focus on the body element when a route transition has occurred. 
+
+In order to not break similar solutions that may already exists in other apps or addons, I propose that the solution be implemented as an optional feature/with a flag. The default could be set to on, and if folks already have a solution in their addons/apps, they could then opt-out of this feature. 
+
+### Phase Two - Intelligent content focus
+
+It would be even better to have a more intelligent solution, since there are several ways new page content could be introduced in an Ember application.
+
+[Ember-self-focused](https://github.com/linkedin/self-focused/tree/master/packages/ember-self-focused) and [ember-a11y](https://github.com/ember-a11y/ember-a11y) are both examples of community-produced addons that attempt to provide a contextual focus, based on new page content. 
+
+From the [Ember-self-focused](https://github.com/linkedin/self-focused/tree/master/packages/ember-self-focused) readme: 
+> When UI transitions happen in a SPA (or in any dynamic web application), there is visual feedback; however, for users of screen reading software, there is no spoken feedback by default. Traditionally, screen reading software automatically reads out the content of a web page during page load or page refresh. In single page applications, the UI transitions happen by rewriting the current page, rather than loading entirely new pages from a server; this makes it difficult for screen reading software users to be aware of UI changes (e.g., clicking on the navigation bar to load a new route).
+
+> If the corresponding HTML node of the dynamic content can be focused programmatically, screen reading software will start speaking the textual content of that node. Focusing the corresponding HTML node of the dynamic content can be considered guided focus management. Not only will it facilitate the announcement of the textual content of the focused HTML node, but it will also serve as a guided context switch. Any subsequent “tab” key press will focus the next focusable element within/after this context. 
+
+The tedious, error-prone task of keeping track of HTML nodes that can/could/would/should have focus is something that Ember could do for you. 
+
+A comparable from another JS Ecosystem: [Reach Router](https://reach.tech/router) is available for React apps. 
+
+## How we teach this
+
+For phase one, the guides should include information about skip links, since the application author will need to add those in themselves (just as they would for a static site). For phase two, we would explain that skip links are no longer needed because we have integrated that functionality in a more machine intelligent way. 
```

Comment:
> For phase two, we would explain that skip links are no longer needed because we have integrated that functionality in a more machine intelligent way. 

Is this true? I usually still keep nav skip links in my SPAs because it's helpful to move past the giant headers / nav bars web apps have.

Reply:
@Robdel12 the idea for phase 2 is that we'd make it so AT would automatically implement skip link functionality so it was a cohesive experience- I think of it as an upgrade to skip links.

Reply:
Skip links are also readily utilized by keyboard-only users, so it would still be a concern outside of ATs. Additionally, as an author, I would expect a means of disabling the generated skip link in case where we want to implement something much more custom and robust.

Reply:
Indeed, AT is a red herring as far as skip-links are concerned.  The main group who _rely_ on skip links are keyboard users _without_ AT.  Screen reader users have a variety of alternative navigation mechanisms available.

Skip-links will still be necessary, until browsers provide at least one built-in mechanism for reaching the main content (e.g. heading navigation, landmark navigation, or spatial navigation).